runtime/thread-pool.cpp

```diff
@@ -0,0 +1,26 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2023 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "runtime/thread-pool.h"
+
+void ThreadPool::init() noexcept {
+  int thread_pool_size = static_cast<int>(std::thread::hardware_concurrency() * thread_pool_ratio);
```

Comment:
let's round up using ceil to ensure it always `> 0` if you specify non-zero ratio